src/components/CreateCourse.js

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from 'react'
+import { ipcRenderer } from 'electron'
+import { hashHistory } from 'react-router'
+
+ipcRenderer.on('add-course', (event, arg) => {
+  console.log('event,arg', arg)
+})
+export default class CreateCourse extends Component {
+  componentWillMount () {
+    this.setState({
+      name: '',
+      startdate: '',
+      duration: null,
+      nameClass: 'inputText',
+      dateClass: 'inputText',
+      durationClass: 'inputText',
+      submitted: false
+    })
+  }
+  sendData () {
+    let obj = {
+      name: this.state.name,
+      duration: this.state.duration,
+      startDate: this.state.startdate
+    }
+    ipcRenderer.send('add-course', obj)
+    this.setState({
+      submitted: true
+    })
+  }
+
+  handleSubmit () {
+    if (this.state.name === '' || this.state.startdate === '' || this.state.duration === null || this.state.duration <= 0) {
+      if (this.state.name === '') {
+        this.setState({nameClass: 'inputText errorInput'})
+      }
+      if (this.state.startdate === '') {
+        this.setState({dateClass: 'inputText errorInput'})
+      }
+      if (this.state.duration === null || this.state.duration <= 0) {
+        this.setState({durationClass: 'inputText errorInput'})
+      }
+    }else {
```

Comment:
The above code is pretty repetitive and a lot of nested ifs. How do you think you could make it cleaner? :)